grafana/resource_api_key.go

```diff
@@ -0,0 +1,118 @@
+package grafana
+
+import (
+	"context"
+	"strconv"
+
+	gapi "github.com/grafana/grafana-api-golang-client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceAPIKey() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAPIKeyCreate,
+		ReadContext:   resourceAPIKeyRead,
+		DeleteContext: resourceAPIKeyDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"role": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"seconds_to_live": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"expiration": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAPIKeyCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	name := d.Get("name").(string)
+	role := d.Get("role").(string)
+	ttl := d.Get("seconds_to_live").(int)
+
+	c := m.(*client).gapi
+	request := gapi.CreateAPIKeyRequest{Name: name, Role: role, SecondsToLive: int64(ttl)}
+	response, err := c.CreateAPIKey(request)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.Set("key", response.Key)
+
+	// Fill the true resource's state after a create by performing a read.
+	resourceAPIKeyRead(ctx, d, m)
+
+	return diags
```

Comment:
You can return the result of the read here. otherwise, read errors go into the void
```suggestion
	// Fill the true resource's state after a create by performing a read
	return resourceAPIKeyRead(ctx, d, m)
```